Add simple dark mode and speed up extract_title() parsing

analogist opened the pull request:
1. Addresses #9 by adding <strike>a simple dark mode inline css block</strike> `bootstrap-dark-5` into the `base.html` template.
2. As `md.parse()` for title extraction (from #11) runs almost as slowly as a full `md.render()`, reduce wasted parsing time by only parsing the first N lines (N=3 default) of each `.md` into tokens for subsequent token search.

analogist committed:
extract_title() examines top N .md lines to minimize parse time

analogist commented:
The dead-simple way of doing dark mode doesn't work -- probably because overwritten by bootstrap css? Will try again later. On the positive side, at least `extract_title()` now no longer noticeably slow the build.

tannewt commented:
Thanks! This might be useful for dark mode: https://vinorodrigues.github.io/bootstrap-dark-5/

analogist committed:
replace bootstrap theme with bootstrap-dark

tannewt commented:
Thanks for working on this! Let me know when you want me to review and merge.

analogist commented:
Thanks, that worked out well as a drop-in! I added SRI to `bootstrap-dark`, and added some debugging output to Actions if the title extraction ever fails. So far, it seems to only do so in unreachable(?) pages like `bill/2021-22/hb/1076/S/rcw/49_labor_regulations/worker_protection_act.md`.

tannewt commented:
> Thanks, that worked out well as a drop-in! I added SRI to `bootstrap-dark`, and added some debugging output to Actions if the title extraction ever fails. So far, it seems to only do so in unreachable(?) pages like `bill/2021-22/hb/1076/S/rcw/49_labor_regulations/worker_protection_act.md`.

Ah, yeah. That looks inaccessible. :-) It's an added chapter. There should be a link from here: https://wa-law.org/bill/2021-22/hb/1076/S/ No need to worry about it now. Those patched RCW pages don't get much traffic yet.

tannewt reviewed (approved):
Thank you!
